src/mbgl/renderer/buckets/raster_bucket.cpp

```diff
@@ -9,12 +9,16 @@ namespace mbgl {
 
 using namespace style;
 
-RasterBucket::RasterBucket(UnassociatedImage&& image_) : image(std::move(image_)) {
+RasterBucket::RasterBucket(UnassociatedImage&& image_) {
+    image = std::make_shared<UnassociatedImage>(std::move(image_));
 }
 
+RasterBucket::RasterBucket(std::shared_ptr<UnassociatedImage> image_): image(image_) {
+
```

Comment:
I've added checks at [line 20](https://github.com/mapbox/mapbox-gl-native/pull/9313/commits/63f85ad056b92426a03ab902cae3292e6a632dbc#diff-5492a713c0e5c0648d604e8d3a6f581fR20) and [line 63](https://github.com/mapbox/mapbox-gl-native/pull/9313/commits/63f85ad056b92426a03ab902cae3292e6a632dbc#diff-5492a713c0e5c0648d604e8d3a6f581fR63).